File: kotlin/math/Pascal.kt
// Print 10 rows of Pascal's triangle, each row zipped from the previous one shifted both ways.
// Run: kotlinc Pascal.kt -include-runtime -d pascal.jar && java -jar pascal.jar

const val ROWS = 10

fun main() {
    var row = listOf(1L)

    repeat(ROWS) {
        println(row.joinToString(" "))
        row = (listOf(0L) + row).zip(row + listOf(0L)) { left, right -> left + right }
    }
}

File: rust/math/pascal.rs
// Print 10 rows of Pascal's triangle, each row zipped from the previous one shifted both ways.
// Build: rustc -o pascal pascal.rs

const ROWS: usize = 10;

fn main() {
    let mut row: Vec<u64> = vec![1];

    for _ in 0..ROWS {
        let fields: Vec<String> = row.iter().map(|value| value.to_string()).collect();
        println!("{}", fields.join(" "));

        let shifted = std::iter::once(&0).chain(row.iter());
        let padded = row.iter().chain(std::iter::once(&0));
        row = shifted.zip(padded).map(|(left, right)| left + right).collect();
    }
}
